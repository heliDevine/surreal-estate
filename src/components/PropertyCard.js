// this component will show properties list
import React from 'react';
import PropTypes from 'prop-types';

const PropertyCard = props => {
    const { title, type, bathrooms, bedrooms, price, city, email } = props;
    return (
        <div className="property-details">
            Property details
            <div className="property-title">title: {title}</div>
            <div className="property-type">type: {type}</div>
            <div className="property-bathrooms">bathrooms: {bathrooms}</div>
            <div className="property-bedrooms">bedrooms: {bedrooms}</div>
            <div className="property-price">price: {price}</div>
            <div className="property-city">city: {city}</div>
            <div className="property-email">city: {email}</div>
        </div>
    );
};

PropertyCard.propTypes = {
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    bathrooms: PropTypes.number.isRequired,
    bedrooms: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    city: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
};

export default PropertyCard;
