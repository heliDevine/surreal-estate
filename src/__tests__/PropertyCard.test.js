import React from 'react';
import { render } from '@testing-library/react';
import PropertyCard from '../components/PropertyCard';

// describe Property card

describe('PropertyCard', () => {
    const validProps = {
        title: 'test',
        type: 'test',
        bathrooms: 2,
        bedrooms: 3,
        price: 140000,
        city: 'Manchester',
        email: 'test@email.com',
    };
    it('renders correctly', () => {
        const { asFragment } = render(
            <PropertyCard
                title={validProps.title}
                type={validProps.type}
                bathrooms={validProps.bathrooms}
                bedrooms={validProps.bedrooms}
                price={validProps.price}
                city={validProps.city}
                email={validProps.email}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correct values to the props', () => {
        const { getByText } = render(
            <PropertyCard
                title={validProps.title}
                type={validProps.type}
                bathrooms={validProps.bathrooms}
                bedrooms={validProps.bedrooms}
                price={validProps.price}
                city={validProps.city}
                email={validProps.email}
            />,
        );
        expect(getByText('test')).toHaveClass('property-title');
        expect(getByText('test')).toHaveClass('property-type');
        expect(getByText('bathrooms')).toHaveClass('property-bathrooms');
        expect(getByText('bedrooms')).toHaveClass('property-bedrooms');
        expect(getByText('price')).toHaveClass('property-price');
        expect(getByText('city')).toHaveClass('property-city');
        expect(getByText('email')).toHaveClass('property-email');
    });
});
