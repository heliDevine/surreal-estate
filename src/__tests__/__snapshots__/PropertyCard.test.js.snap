// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-details"
  >
    Property details
    <div
      class="property-title"
    >
      title: test
    </div>
    <div
      class="property-type"
    >
      type: test
    </div>
    <div
      class="property-bathrooms"
    >
      bathrooms: 2
    </div>
    <div
      class="property-bedrooms"
    >
      bedrooms: 3
    </div>
    <div
      class="property-price"
    >
      price: 140000
    </div>
    <div
      class="property-city"
    >
      city: Manchester
    </div>
    <div
      class="property-email"
    >
      city: test@email.com
    </div>
  </div>
</DocumentFragment>
`;
